modules/statistics/php/stats_behavioural.class.inc

```diff
@@ -72,82 +77,97 @@ class Stats_Behavioural extends \NDB_Form
+        $centers = $DB->pselect(
+            "SELECT CONCAT('C', CenterID) as ID,
+            CenterID as NumericID,
+            IFNULL(PSCArea,Name) as LongName,
+            Name as ShortName
+            FROM psc
+            WHERE CenterID <> '1'
+            AND Study_site = 'Y'",
+            array()
+        );
+        if (empty($centers)) {
+            error_log('No center data could be retrieved from the database!');
+        } else {
+            foreach ($centers as $row) {
+                $id        = $row['NumericID'] ?? '';
+                $shortName = $row['ShortName'] ?? '';
+                $longName  = $row['LongName'] ?? '';
+                $this->tpl_data['Sites'][$id]
+                    = $longName;
+                if ($site == $id) {
+                    $this->tpl_data['CurrentSite'] = array(
+                                                      'ID'   => $id,
+                                                      'Name' => $longName,
+                                                     );
+                }
+            }
+            $this->tpl_data['Centers'] = $centers;
         }
 
-        if ($_REQUEST['BehaviouralProject'] ?? ''
-        ) {
-            $currentProject = htmlspecialchars($_REQUEST['BehaviouralProject']);
+        $Param_Project      = '';
+        $subproject_query   = '';
+        $behaviouralProject = $_REQUEST['BehaviouralProject'] ?? '';
+        if (empty($behaviouralProject)) {
+            error_log('No behavioural project specified in request');
+        } else {
+            $behaviouralProject = htmlspecialchars($behaviouralProject);
             $this->tpl_data['CurrentProject']
                 = array(
-                   'ID'   => $currentProject,
-                   'Name' => $projects[$currentProject],
+                   'ID'   => $behaviouralProject,
+                   'Name' => $projects[$behaviouralProject],
                   );
             $Param_Project       ='AND (c.ProjectID IS NULL OR c.ProjectID=:pid) ';
-            $this->params['pid'] = htmlspecialchars($_REQUEST['BehaviouralProject']);
-        } else {
-            $Param_Project = '';
-        }
-        $subprojects = \Utility::getSubprojectsForProject($currentProject);
+            $this->params['pid'] = $behaviouralProject;
+            $subprojects         = \Utility::getSubprojectsForProject(
+                $behaviouralProject
+            );
 
-        $this->tpl_data['Subprojects'] = $subprojects;
-        foreach ($subprojects as $key=>$val) {
-            $subprojList .= $key.",";
-        }
-        $subprojList = substr($subprojList, 0, -1);
-        if (!empty($subprojList)) {
-            $suproject_query ="AND s.SubprojectID IN ($subprojList)";
-        } else {
-            $suproject_query ='';
+            $this->tpl_data['Subprojects'] = $subprojects;
+            foreach ($subprojects as $key=>$val) {
+                $subprojList .= $key.",";
+            }
```

Comment:
Could use implode(array_keys()) instead of foreach(). This would prevent creating the unused var $val